src/accelerate/accelerator.py

```diff
@@ -221,7 +221,9 @@ def __init__(
             )
             mixed_precision = "fp16"
 
-        if dynamo_backend is not None:
+        if dynamo_backend is None:
+            dynamo_backend = DynamoBackend(os.environ.get("DYNAMO_BACKEND", DynamoBackend.NO))
```

Comment:
This is done in the state [here](https://github.com/huggingface/accelerate/blob/8a96b0bfb856f37cf5013384454aaecc680ca23d/src/accelerate/state.py#L79) so I don't understand why this line is necessary.